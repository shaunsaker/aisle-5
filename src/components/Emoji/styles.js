import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  emoji: {
    fontSize: styleConstants.fonts.sizes.emoji,
  },
});

export default styles;
