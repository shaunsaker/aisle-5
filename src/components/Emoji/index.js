import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import emoji from 'node-emoji';

import styles from './styles';

const propTypes = {
  name: PropTypes.string,
  size: PropTypes.number,
};

const defaultProps = {};

/*
 Returns an emoji if there is a match with the name
*/
const Emoji = ({ name, size }) => {
  const emojiMatches = name && emoji.search(name.toLowerCase());
  const emojiMatch = name && emojiMatches[0];
  const emojiKey = emojiMatch && emojiMatch.key;
  const actualEmoji = emojiKey && emoji.get(emojiKey);

  if (actualEmoji) {
    return <Text style={[styles.emoji, { fontSize: size }]}>{actualEmoji} </Text>;
  }

  return null;
};

Emoji.propTypes = propTypes;
Emoji.defaultProps = defaultProps;

export default Emoji;
