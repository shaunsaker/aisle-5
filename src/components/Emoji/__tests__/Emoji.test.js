import React from 'react';
import renderer from 'react-test-renderer';

import Emoji from '..';

describe('Emoji', () => {
  const name = 'Beer';

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Emoji />);

      expect(component).toMatchSnapshot();
    });

    it('renders the default state', () => {
      const component = renderer.create(<Emoji name={name} />);

      expect(component).toMatchSnapshot();
    });
  });
});
