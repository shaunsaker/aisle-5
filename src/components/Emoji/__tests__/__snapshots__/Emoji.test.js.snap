// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji renders renders the default state 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 20,
      },
      Object {
        "fontSize": undefined,
      },
    ]
  }
>
  🍺
   
</Text>
`;

exports[`Emoji renders renders with minimum required props 1`] = `null`;
